### ISSUE TITLE
"COLLATE utf8_general_ci " breaks Postgresql 

### ISSUE THREAD

**Issue by Carl4** [cortezaproject/corteza-server#140]
https://github.com/cortezaproject/corteza-server/blob/e6bdf69defe337a8a0ca09a07c6e19d8e2c15722/pkg/ql/squirrel.go#L106

It looks like `utf8_general_ci` is a MySQL-specific Collation name.  Postgres uses names like `en_US`.

I'm looking into whether it's possible to simply rename the `en_US.utf8` Collation in Postgres. But for now (in release 2020.12), text search doesn't work with a postgresql backend.

SQL Errors look like so: 

`STATEMENT:  SELECT crd.id, crd.module_id, crd.rel_namespace, crd.owned_by, crd.created_by, crd.updated_by, crd.deleted_by, crd.created_at, crd.updated_at, crd.deleted_at FROM compose_record AS crd LEFT JOIN compose_record_value AS rv_ContactName ON (rv_ContactName.record_id = crd.id AND rv_ContactName.name = 'ContactName' AND rv_ContactName.deleted_at IS NULL) LEFT JOIN compose_record_value AS rv_Number ON (rv_Number.record_id = crd.id AND rv_Number.name = 'Number' AND rv_Number.deleted_at IS NULL) WHERE crd.module_id = $1 AND crd.rel_namespace = $2 AND crd.deleted_at IS NULL AND (rv_ContactName.value  COLLATE utf8_general_ci LIKE $3 OR rv_Number.value  COLLATE utf8_general_ci LIKE $4) ORDER BY rv_Number.value  ASC NULLS FIRST, id ASC NULLS FIRST LIMIT 11
2021-02-22 19:04:28.066 GMT [875] ERROR:  collation "utf8_general_ci" for encoding "UTF8" does not exist at character 599`

**Comment by Carl4** [comment]
Took a little digging, but it's apparently possible to [copy postgres collations](https://www.postgresql.org/docs/11/sql-createcollation.html).  Running the following on my postgres database corrected the errors by creating a collation with the same name as MySQL uses.

```
corteza=> CREATE COLLATION utf8_general_ci FROM "en_US";
CREATE COLLATION
```

**Comment by darh** [comment]
Thanks for digging deep enough to find the solution @Carl4.
This will have to be for now until we find a better way change SQL flavour for different databases at this level.

**Comment by darh** [comment]
Was fixed and backported to `2020.12.x` branch. Will be part of `2020.12.4` release
